sktime/transformers/series_as_features/dwt.py

```diff
@@ -0,0 +1,124 @@
+import numpy as np
+import pandas as pd
+import math
+from sktime.transformers.series_as_features.base import \
+    BaseSeriesAsFeaturesTransformer
+from sktime.utils.data_container import tabularize
+from sktime.utils.validation.series_as_features import check_X
+
+__author__ = "Vincent Nicholson"
+
+
+class DWT(BaseSeriesAsFeaturesTransformer):
+
+    def __init__(self, num_levels=3):
+        self.num_levels = num_levels
+        super(DWT, self).__init__()
+
+    """
+    Parameters
+    ----------
+    X : a pandas dataframe of shape = [n_samples, num_dims]
+        The training input samples.
+
+    Returns
+    -------
+    dims: a pandas data frame of shape = [n_samples, num_dims]
+    """
+    def transform(self, X, y=None):
+
+        # Check the data
+        self.check_is_fitted()
+        X = check_X(X, enforce_univariate=False)
+
+        # Get information about the dataframe
+        col_names = X.columns
+
+        df = pd.DataFrame()
+        for x in col_names:
+            # Convert one of the columns in the dataframe to numpy array
+            arr = tabularize(pd.DataFrame(X[x]), return_array=True)
+
+            transformedData = self.extract_wavelet_coefficients(arr)
+
+            # Convert to a numpy array
+            transformedData = np.asarray(transformedData)
+
+            # Add it to the dataframe
+            colToAdd = []
+            for i in range(len(transformedData)):
+                inst = transformedData[i]
+                colToAdd.append(pd.Series(inst))
+
+            df[x] = colToAdd
+
+        return df
+
+    """
+    Function to extract the wavelet coefficients of a 2d array of time series.
+
+    The coefficients correspond to the wavelet coefficients from levels 1 to
+    num_levels followed by the approximation coefficients of the highest level.
+    """
+    def extract_wavelet_coefficients(self, data):
```

Comment:
I'd make this private by using the leading underscore convention

```suggestion
    def _extract_wavelet_coefficients(self, data):
```